File: src/utils/math.ts
export function clamp(num: number, min: number, max: number): number {
    if (num <= min) {
        return min;
    }
    return num >= max ? max : num;
}

export function roundToPrecision(value: number, precision = 0): number {
    if (!Number.isNaN(Number(value))) {
        return Math.round(value * (10 ** precision)) / (10 ** precision);
    }
    return 0;
}

File: src/commonMain/kotlin/com/equisoft/taxca/utils/Math.kt
package com.equisoft.taxca.utils

import kotlin.math.floor
import kotlin.math.pow

fun clamp(num: Double, min: Double, max: Double): Double {
    if (num <= min) {
        return min
    }
    return if (num >= max) max else num
}

fun roundToPrecision(value: Double, precision: Int = 0): Double {
    if (value.isNaN()) {
        return 0.0
    }
    val factor = 10.0.pow(precision)
    // floor(x + 0.5) matches JS Math.round (ties toward +Infinity), unlike
    // kotlin.math.round which rounds ties away from zero for negative values.
    return floor(value * factor + 0.5) / factor
}
